vm/systemSmartContracts/esdt.go

```diff
@@ -409,16 +409,20 @@ func isTokenNameHumanReadable(tokenName []byte) bool {
 
 func (e *esdt) createNewTokenIdentifier(caller []byte, ticker []byte) ([]byte, error) {
 	newRandomBase := append(caller, e.eei.BlockChainHook().CurrentRandomSeed()...)
-	newRandom := sha256.Sha256{}.Compute(string(newRandomBase))
+	newRandom := e.hasher.Compute(string(newRandomBase))
 	newRandomForTicker := newRandom[:tickerRandomSequenceLength]
 
 	tickerPrefix := append(ticker, []byte(tickerSeparator)...)
 	newRandomAsBigInt := big.NewInt(0).SetBytes(newRandomForTicker)
 
 	one := big.NewInt(1)
 	for i := 0; i < numOfRetriesForIdentifier; i++ {
-		encoded := hex.EncodeToString(newRandomAsBigInt.Bytes())
-		newIdentifier := append(tickerPrefix, []byte(encoded)...)
+		encoded := []byte(hex.EncodeToString(newRandomAsBigInt.Bytes()))
+		lenEncoded := len(encoded)
+		if lenEncoded < encodedRandomSequenceLength {
+			encoded = append(make([]byte, encodedRandomSequenceLength-lenEncoded), encoded...)
```

Comment:
wouldn't it make more sense to prepend the character '0' instead of \x00 bytes?

Reply:
:+1:

Reply:
`fmt.Sprintf("%06x", big.NewInt(123))` does it in one line, the only problem is I don't think you can provide the number of digits as a variable. But I think encodedRandomSequenceLength is a constant anyway, up to you to decide

Reply:
...or you could use bytes.Repeat function

Reply:
done as @andrei-marinica mentioned.